base/compiler/abstractinterpretation.jl

```diff
@@ -42,9 +42,12 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
     fullmatch = Bool[]
     if splitunions
         splitsigs = switchtupleunion(atype)
```

Comment:
```suggestion
```

The `unionsplitcost(atype_params)` would also potentially be `unionsplitcost(argtypes)` (I assume no lattice element can become a union after widenconst if it wasn't before)

Reply:
AFAIU no "constant" lattice element wraps `Union` type, so this `unionsplitcost(argtypes)` should be valid.
I added a note on this. Does this makes sense and do you think it's worth to be left ?
https://github.com/JuliaLang/julia/blob/e5a51a4cb4b19b388c056cd75a42d04b31d05213/base/compiler/abstractinterpretation.jl#L39-L40